homeassistant/components/incomfort/binary_sensor.py

```diff
@@ -21,10 +21,12 @@
 class IncomfortFailed(BinarySensorDevice):
     """Representation of an InComfort Failed sensor."""
 
-    def __init__(self, client, boiler) -> None:
+    def __init__(self, client, heater) -> None:
         """Initialize the binary sensor."""
+        self._unique_id = f"{DOMAIN}_{heater.serial_no}_failed"
```

Comment:
We don't need to include the domain in the unique_id. The entity registry is aware of both the integration domain and the platform domain of entities.

Reply:
OK, removed all the `{DOMAIN}_`s.